src/initio/initio.c

```diff
@@ -0,0 +1,848 @@
+    }
+
+    if (dir != -1) {
+        if (mraa_gpio_dir(dev, (mraa_gpio_dir_t) dir) != MRAA_SUCCESS) {
+            syslog(LOG_ERR, "parse_gpio: error setting up gpio direction %s", proto[idx]);
+            mraa_gpio_close(dev);
+            return NULL;
+        } else {
+            if (++idx == n) {
+                return dev;
+            }
+        }
+    } else {
+        // Set direction to default - output
+        if (mraa_gpio_dir(dev, MRAA_GPIO_OUT) != MRAA_SUCCESS) {
+            syslog(LOG_ERR, "parse_gpio: error setting up gpio direction %s", G_DIR_OUT);
+            mraa_gpio_close(dev);
+            return NULL;
+        }
+    }
+
+    // Check the value.
+    int value = -1;
+    if (mraa_atoi_x(proto[idx], NULL, &value, 0) == MRAA_SUCCESS) {
+        if (mraa_gpio_write(dev, value) != MRAA_SUCCESS) {
+            syslog(LOG_ERR, "parse_gpio: could not init gpio number %d with value %d", gpio_num, value);
+            mraa_gpio_close(dev);
+            return NULL;
+        } else {
+            if (++idx == n) {
+                return dev;
+            }
+        }
+    }
+
+    /* Check for mode. */
+    int mode = -1;
+    if (strncmp(proto[idx], G_MODE_STRONG, strlen(G_MODE_STRONG)) == 0) {
+        mode = MRAA_GPIO_STRONG;
+    } else if (strncmp(proto[idx], G_MODE_PULLUP, strlen(G_MODE_PULLUP)) == 0) {
+        mode = MRAA_GPIO_PULLUP;
+    } else if (strncmp(proto[idx], G_MODE_PULLDOWN, strlen(G_MODE_PULLDOWN)) == 0) {
+        mode = MRAA_GPIO_PULLDOWN;
+    } else if (strncmp(proto[idx], G_MODE_HIZ, strlen(G_MODE_HIZ)) == 0) {
+        mode = MRAA_GPIO_HIZ;
+    } else if (strncmp(proto[idx], G_MODE_ACTIVE_LOW, strlen(G_MODE_ACTIVE_LOW)) == 0) {
+        mode = MRAA_GPIOD_ACTIVE_LOW;
+    } else if (strncmp(proto[idx], G_MODE_OPEN_DRAIN, strlen(G_MODE_OPEN_DRAIN)) == 0) {
+        mode = MRAA_GPIOD_OPEN_DRAIN;
+    } else if (strncmp(proto[idx], G_MODE_OPEN_SOURCE, strlen(G_MODE_OPEN_SOURCE)) == 0) {
+        mode = MRAA_GPIOD_OPEN_SOURCE;
+    }
+
+    if (mode != -1) {
+        if (mraa_gpio_mode(dev, (mraa_gpio_mode_t) mode) != MRAA_SUCCESS) {
+            syslog(LOG_ERR, "parse_gpio: error setting up gpio mode %s", proto[idx]);
+            mraa_gpio_close(dev);
+            return NULL;
+        } else {
+            if (++idx == n) {
+                return dev;
+            }
+        }
+    }
+
+    /* Check for edge. */
+    int edge = -1;
+    if (strncmp(proto[idx], G_EDGE_NONE, strlen(G_EDGE_NONE)) == 0) {
+        edge = MRAA_GPIO_EDGE_NONE;
+    } else if (strncmp(proto[idx], G_EDGE_BOTH, strlen(G_EDGE_BOTH)) == 0) {
+        edge = MRAA_GPIO_EDGE_BOTH;
+    } else if (strncmp(proto[idx], G_EDGE_RISING, strlen(G_EDGE_RISING)) == 0) {
+        edge = MRAA_GPIO_EDGE_RISING;
+    } else if (strncmp(proto[idx], G_EDGE_FALLING, strlen(G_EDGE_FALLING)) == 0) {
+        edge = MRAA_GPIO_EDGE_FALLING;
+    }
+
+    if (edge != -1) {
+        if (mraa_gpio_dir(dev, (mraa_gpio_edge_t) edge) != MRAA_SUCCESS) {
```

Comment:
Double check here, this seems incorrect.

Reply:
fixed